tests/linalg_test.py

```diff
@@ -134,6 +134,51 @@ def norm(x):
     self._CompileAndCheck(partial(np.linalg.eigh, UPLO=uplo), args_maker,
                           check_dtypes=True)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_n={}_full_matrices={}_compute_uv={}".format(
+          jtu.format_shape_dtype_string((m, n), dtype), full_matrices, compute_uv),
+       "m": m, "n": n, "dtype": dtype, "full_matrices": full_matrices,
+       "compute_uv": compute_uv, "rng": rng}
+      for m in [2, 7, 29, 53]
+      for n in [2, 7, 29, 53]
+      for dtype in float_types()
+      for full_matrices in [False, True]
+      for compute_uv in [False, True]
+      for rng in [jtu.rand_default()]))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSVD(self, m, n, dtype, full_matrices, compute_uv, rng):
+    if not hasattr(lapack, "jax_gesdd"):
+      self.skipTest("No singular value decomposition implementation available")
+    args_maker = lambda: [rng((m, n), dtype)]
+
+    # Norm, adjusted for dimension and type.
+    def norm(x):
+      n = onp.linalg.norm(x, axis=(-2, -1))
+      return n / (max(m, n) * onp.finfo(dtype).eps)
+
+    a, = args_maker()
+    out = np.linalg.svd(a, full_matrices=full_matrices, compute_uv=compute_uv)
+
+    if compute_uv:
```

Comment:
It appears we are testing nothing if compute_uv is false. I think the singular values are unique up to reordering; could we perhaps compare them against those derived from onp.linalg.svd?

Reply:
Sure, I will test this as well. Sorry about missing it.